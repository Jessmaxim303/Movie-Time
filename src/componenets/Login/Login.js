import React, { Component } from 'react';
import { userSignIn, fetchRatingsAPI } from '../../apiCalls/apiCalls'
import { connect } from 'react-redux';
import { addUser, errorMsg, getRatings } from '../../actions';
import { Redirect } from 'react-router-dom';
import './Login.scss';
import PropTypes from 'prop-types';

export class Login extends Component {
  constructor(props) {
    super(props)
    this.state = {
      "email": "",
      "password": ""
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value})
  }

  handleClick = async () => {
    const { addUser } = this.props;
    const { email, password } = this.state;
    const user = { email, password };
    try {
      const result = await userSignIn(user);
      addUser(result.user)
      this.props.errorMsg('')
      this.loadRatings()
    }
    catch (e) {
      this.props.errorMsg('Username and/or Password are incorrect')
    }
  }

  loadRatings() {
    fetchRatingsAPI(this.props.user.id)
    .then(data => {
      this.props.getRatings(data.ratings)
    })
  }

  render() {
    const { email, password } = this.state
    return (
      <div className='login-form'>
        <input
          className='login-form__email'
          type="text"
          placeholder="Email..."
          name="email"
          value={email}
          onChange={this.handleChange}
        />
        <input
          className='login-form__password'
          type="password"
          placeholder="Password..."
          name="password"
          value={password}
          onChange={this.handleChange}
        />
        <p className='login-form__error'>{this.props.error}</p>
        <button className='login-form__button' onClick={() => this.handleClick()}>Login</button>
        {this.props.user ? <Redirect to='/' /> : null}
    </div>)
  }
}

export const mapStateToProps = state => ({
  error: state.error,
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  addUser: (user) => dispatch( addUser(user) ),
  errorMsg: error => dispatch( errorMsg(error) ),
  getRatings: (ratings) => dispatch(getRatings(ratings))
})

export default connect(mapStateToProps, mapDispatchToProps)(Login)

Login.propTypes = {
  props: PropTypes.any
};
