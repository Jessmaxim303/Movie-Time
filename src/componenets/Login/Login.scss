@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.login-form {
  @include display-flex (column nowrap, center, center);
  border: 0.25em $primary-color solid;
  flex: 0.5;
  justify-self: center;
  align-self: center;
  border-radius: 1em;
  background-color: $card-background;
  padding: 1.5em;
  margin: 2em;
  width: 30%;
  &__email,
  &__password {
    font-family: $main-font;
    font-size: 1.5em;
    margin: 1em 0 0.75em;
  }
  &__button{
    @include button-style;
    margin: 0.25em 0 0.75em;
    width: 100%;
    &:hover {
      transform: none;
      background-color: darken($primary-color, 10);
    }
  }
  &__error {
    color: red;
    }
}