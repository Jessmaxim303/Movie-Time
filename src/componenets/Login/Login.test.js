import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login.js';

describe('Login', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Login />)
  })
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})