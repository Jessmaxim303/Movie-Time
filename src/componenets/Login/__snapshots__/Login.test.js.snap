// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match a snapshot 1`] = `
<div
  className="login-form"
>
  <input
    className="login-form__email"
    name="email"
    onChange={[Function]}
    placeholder="Email..."
    type="text"
    value=""
  />
  <input
    className="login-form__password"
    name="password"
    onChange={[Function]}
    placeholder="Password..."
    type="password"
    value=""
  />
  <p
    className="login-form__error"
  />
  <button
    className="login-form__button"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
